packages/realm-react/src/useObject.tsx

```diff
@@ -29,6 +29,7 @@ import { CollectionCallback, getObjectForPrimaryKey, getObjects } from "./helper
 export function createUseObject(useRealm: () => Realm) {
   function useObject<T>(type: string, primaryKey: T[keyof T]): (T & Realm.Object<T>) | null;
   function useObject<T extends Realm.Object>(type: { new (...args: any): T }, primaryKey: T[keyof T]): T | null;
+  function useObject<T extends Realm.Object<T>>(type: { new (...args: any): T }, primaryKey: T[keyof T]): T | null;
```

Comment:
I'd say it's simpler to change the overload above to `<T extends AnyRealmObject>` (which would be equivalent of `<T extends Realm.Object<any>>`)?

Reply:
That's what `Realm.objects` does anyway: https://github.com/realm/realm-js/blob/b7b4c6f7cb276ab47acf59ed2b263cd28bbd3bfa/packages/realm/src/Realm.ts#L982

Reply:
Yes, that is a better way.  Thanks!